lib/usecases/get_survey_detail_use_case.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter/api/repository/survey_repository.dart';
+import 'package:survey_flutter/model/response/survey_detail_data_response.dart';
+import 'package:survey_flutter/usecases/base/base_use_case.dart';
+
+final getSurveyDetailUseCaseProvider = Provider(
+    (ref) => GetSurveyDetailUseCase(ref.watch(surveyRepositoryProvider)));
+
+class GetSurveyDetailUseCase extends UseCase<SurveyDetailDataResponse, String> {
+  final SurveyRepository _surveyRepository;
+
+  GetSurveyDetailUseCase(this._surveyRepository);
+
+  @override
+  Future<Result<SurveyDetailDataResponse>> call(String params) async {
```

Comment:
we should define a specific name instead of a general name like `params`
```suggestion
  Future<Result<SurveyDetailDataResponse>> call(String surveyId) async {
```

Reply:
@manh-t I tried to rename it before but I got `The parameter name 'surveyId' doesn't match the name 'params' in the overridden method` error. That's why I left them like that. Is there any other way around that? 💭

Reply:
@kaungkhantsoe it's not an error, just a warning and we can bypass that warning

Reply:
Fixed in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/81/commits/bfc579d37d29ab990e9509f9bc0002cfe653c89a